### PR TITLE
MRG: refactor internals for epochs image plot

### PR DESCRIPTION
**Pull request by jona-sassenhagen**
This is to make it easier to handle the unification of topo plotters and single sensor plotters.

@drammock

### CODE REVIEW

**Comment by jona-sassenhagen** [comment]
Sketch:

For epochs.topo_topo,
    - create data array(s; for multiple types)
    - apply order, filter, vlim funcs in mne.viz.epochs.py
    - and off to the topo plotter

@drammock

**Comment by jona-sassenhagen** [comment]
@drammock this too ...

**Comment by jona-sassenhagen** [comment]
@drammock I have moved epochs objects out of most of the helper functions and made everything a bit more modular. This should now be more conductive to looping over axes and data and calling functions to handle and eventually plot arrays.

**Comment by larsoner** [comment]
But is it MRG?????

**Comment by jona-sassenhagen** [comment]
no :(

**Comment by jona-sassenhagen** [comment]
actually yes

File: mne/viz/epochs.py [hunk 1]
```diff
@@ -428,27 +426,47 @@ def _prepare_epochs_image_im_data(epochs, ch_type, overlay_times, order,
...
+def _prepare_epochs_image_im_data(data, times, ch_type, overlay_times, order,
+                                  sigma, vmin, vmax, scaling, ts_args):
+    """Preprocess epochs image (sort, filter). Helper for plot_epochs_image."""
+    from scipy import ndimage
+
+    # data transforms - sorting, scaling, smoothing
+    data, overlay_times = _order_epochs(data, times, order, overlay_times)
+
+    if sigma > 0.:
+        data = ndimage.gaussian_filter1d(data, sigma=sigma, axis=0)
```
Comment (larsoner) [inline mne/viz/epochs.py]: It is not obvious to me which axis this is smoothing. I assume 0 is still epochs, so for each channel and time point it's smoothing across epochs / repetitions (?)

Comment (larsoner) [reply mne/viz/epochs.py]: (In other words, please add a comment)

Otherwise still LGTM

**Comment by larsoner** [comment]
Thanks @jona-sassenhagen

**Pull request merged**